### PR TITLE
code-review-checklists

### PR DESCRIPTION
**Pull request by luizaandrade**
Now on the right repo

### CODE REVIEW

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Data cleaning code review checklist
+
+**Data source/survey round**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:3]:
```suggestion
**Data source/survey round:**
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+**Date**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:5]:
```suggestion
**Date:**  
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Data cleaning code review checklist
+
+**Data source/survey round**
+
+**Date**  
+
+## List of files to be checked [Add names or links]
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:7]:
```suggestion
## List of files to be checked 
_Add names or links_:
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- Master script
+- Analysis scripts (list by order of priority)
+- List of outputs
+
+## List of auxiliary documents
+- Documentation
+- Originally created outputs
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:14]:
```suggestion
- Originally-created outputs
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Reproducibility
+- [ ] All scripts run from the master after adding the correct folder path to line(s) X (and XX)
+- [ ] The master script is organized in a way that allows you to understand the general tasks being performed in the code
+- [ ] The master script tracks which scripts create and use which files
+- [ ] All the outputs are recreated and are exactly the same as the ones shared by the coder
+
+## Code organization and readability
+- [ ] Code names are informative
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:23]:
```suggestion
- [ ] Code file names are informative and/or link directly to outputs
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- [ ] It is clear in the code why tasks are being executed
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Analysis scripts
+- [ ] No variables are created in analysis scripts unless very specific to a particular output
+- [ ] Code is modular
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:33]:
```suggestion
- [ ] Code is modular, such that individual outputs can be run in any order
```

File: dime-coding-standards/checklists/analysis-code-review.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+- [ ] Research decisions are well documented in the code (e.g. sample used, treatment of missing variables)
+- [ ] The code implements the models described in the documentation
+- [ ] The functions used are apropriate for the analysis being performed
+- [ ] Categorical variables are used correctly (i.e., labeled integers are not used as continuous variables)
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/analysis-code-review.md:37]:
```suggestion
- [ ] Categorical variables are used correctly (i.e. labeled integers are not used as continuous variables)
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
+# Data cleaning code review checklist
+
+**Data source/survey round**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:3]:
```suggestion
**Data source/survey round:**
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+
+**Date**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:5]:
```suggestion
**Date:**  
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+**Date**  
+
+## List of files to be checked [Add names or links]
+- Master script
+- Constructed dataset(s)
+- Construction scripts (list by order of priority)
+
+## List of auxiliary documents
+- Documentation
+- Main variables to be checked
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:14]:
```suggestion
- Main constructed variables to be checked
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 11]
```diff
@@ -0,0 +1,44 @@
+# Data cleaning code review checklist
+
+**Data source/survey round**
+
+**Date**  
+
+## List of files to be checked [Add names or links]
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:7]:
```suggestion
## List of files to be checked
_Add names or links_:
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+- Documentation
+- Main variables to be checked
+
+## Reproducibility
+- [ ] All scripts run from the master after adding the correct folder path to line(s) X (and XX)
+- [ ] The master script is organized in a way that allows you to understand the general tasks being performed in the code
+- [ ] The master script tracks which scripts create and use which files
+
+## Code organization and readability
+- [ ] Code names are informative
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:22]:
```suggestion
- [ ] Code file names are informative 
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] It is clear in the code why tasks are being executed
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Construction scripts
+- [ ] Do you understand how and why each variable is constructed? If not, indicate to the author where more comments are needed.
+- [ ] Check merges. Are any observations dropped? If so, is there a clear justification for that? If any observations didn't match, is that explained in the comments?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:32]:
```suggestion
- [ ] Check merges. Are any observations dropped or does the number of observations increase? If so, is there a clear justification for that? If any observations didn't match, is that explained in the comments?
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Check collapses. How are missing values treated?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:33]:
```suggestion
- [ ] Check collapses, reshapes, and groupwise calculation (`bys` and `even` for example). How are missing values treated? Does the sort order of the data matter or is the result uniquely determined? Is the number of observations correct and are the mathematics correct?
```

Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:33]:
```suggestion
- [ ] Check collapses, reshapes, and groupwise calculations (`bys` and `egen` for example). How are missing values treated? Does the sort order of the data matter or is the result uniquely determined? Is the number of observations correct and are the mathematics correct?
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 15]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Check winsorization. Is the reason for winsorizing explained? Is there documentation explaning how the percentile was chosen and why one/both tails were winsorized?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:34]:
```suggestion
- [ ] Check winsorization or other techniques for handling outliers. Is the reason for the chosen transformation explained? Is there documentation explaining how parameters such as cutoff percentiles were chosen and why one or both tails were altered?
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Check treatment of missing values. Are research decisions well documented? Are there cases where you think missing values are being created or replaced unintentionally?
+- [ ] Check creation of new variables. Does the code match the variable definition in the documentation? Is the correct function being used?
+- [ ] Is there always clear documentation of why observations are dropped, if any?
+
+## Constructed dataset(s)
+- [ ] Is the resulting data set tidy (each row is an observation, each column is a variable)?
+- [ ] Are variable names informative?
+- [ ] Are variable labels informative?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:42]:
```suggestion
- [ ] Are variable labels informative?
- [ ] Are value labels informative?
- [ ] Are all labels grammatically correct and free of special characters?
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Is there clear documentation (variable dictionary, variable labels, notes, comments) about variable definition?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:43]:
```suggestion
- [ ] Is there clear documentation (variable dictionary, variable labels, value labels, notes, comments) about variable definition?
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Are data set names informative?
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/construction-code-review.md:44]:
```suggestion
- [ ] Are data set file names informative?
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
+# Data cleaning code review checklist
+
+**Data source/survey round**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:3]:
```suggestion
**Data source/survey round:**
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+
+**Date**
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:5]:
```suggestion
**Date:**  
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+- Cleaning scripts
+
+## Reproducibility
+- [ ] All scripts run from the master after adding the correct folder path to line(s) X (and XX)
+- [ ] The master script is organized in a way that allows you to understand the general tasks being performed in the code
+- [ ] The master script tracks which scripts create and use which files
+- [ ] The data sets created by the reviewer are exactly the same as those shared by the coder
+
+## Code organization and readability
+- [ ] Code names are informative
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:19]:
```suggestion
- [ ] Code file names are informative 
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 22]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] It is clear in the code why tasks are being executed
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Clean data set checks (pre-publication)
+- [ ] The data does not include direct identifiers
+- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:29]:
```suggestion
- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable (or variables)
```

Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:29]:
```suggestion
- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable(s)
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] The level of observation of the data set is clear from the dataset name, ID variables and documentation
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:30]:
```suggestion
- [ ] The level of observation of the data set is clear from the dataset name, ID variables, and documentation
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] Variables have informative labels or an acompanying dictionary
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:31]:
```suggestion
- [ ] Variables have informative labels or an accompanying dictionary
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] Categorical variables have clear and informative value labels
+- [ ] No modification is made from the raw to the clean data other then correcting problems
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:33]:
```suggestion
- [ ] No modification is made from the raw to the clean data other than correcting problems 
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 26]
```diff
@@ -0,0 +1,41 @@
...
+## Clean data set checks (pre-publication)
+- [ ] The data does not include direct identifiers
+- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable
+- [ ] The level of observation of the data set is clear from the dataset name, ID variables and documentation
+- [ ] Variables have informative labels or an acompanying dictionary
+- [ ] Categorical variables have clear and informative value labels
+- [ ] No modification is made from the raw to the clean data other then correcting problems 
+- [ ] No raw variables are processed (winsorized, for example)  
+- [ ] Variables can be easily traced back to the original questionnaire
+
```
Comment (bbdaniels) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:36]: - [ ] Variables are sensibly ordered and dataset is compressed and saved in appropriate format

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 27]
```diff
@@ -0,0 +1,41 @@
...
+## Clean data set checks (pre-publication)
+- [ ] The data does not include direct identifiers
+- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable
+- [ ] The level of observation of the data set is clear from the dataset name, ID variables and documentation
+- [ ] Variables have informative labels or an acompanying dictionary
+- [ ] Categorical variables have clear and informative value labels
+- [ ] No modification is made from the raw to the clean data other then correcting problems 
+- [ ] No raw variables are processed (winsorized, for example)  
+- [ ] Variables can be easily traced back to the original questionnaire
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:35]:
```suggestion
- [ ] Variables can be easily traced back to the original questionnaire
- [ ] Variables are sensibly ordered and dataset is compressed and saved in appropriate format
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+- [ ] Do you understand how and why each variable is constructed? If not, indicate to the author where more comments are needed.
+- [ ] Check merges. Are any observations dropped? If so, is there a clear justification for that? If any observations didn't match, is that explained in the comments?
+- [ ] Check collapses. How are missing values treated?
+- [ ] Check winsorization. Is the reason for winsorizing explained? Is there documentation explaning how the percentile was chosen and why one/both tails were winsorized?
+- [ ] Check treatment of missing values. Are research decisions well documented? Are there cases where you think missing values are being created or replaced unintentionally?
+- [ ] Check creation of new variables. Does the code match the variable definition in the documentation? Is the correct function being used?
+- [ ] Is there always clear documentation of why observations are dropped, if any?
+
+## Constructed dataset(s)
+- [ ] Is the resulting data set tidy (each row is an observation, each column is a variable)?
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/construction-code-review.md:41]:
```suggestion
- [ ] Is the resulting dataset tidy (each row is an observation, each column is a variable)?
```

File: dime-coding-standards/checklists/construction-code-review.md (hunk 18 above)
Comment (luizaandrade) [inline dime-coding-standards/checklists/construction-code-review.md]:
```suggestion
- [ ] Are dataset names informative?
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 29]
```diff
@@ -0,0 +1,41 @@
...
+## List of files to be checked [Add names or links]
+- Master script
+- Clean dataset(s)
+- Cleaning scripts
+
+## Reproducibility
+- [ ] All scripts run from the master after adding the correct folder path to line(s) X (and XX)
+- [ ] The master script is organized in a way that allows you to understand the general tasks being performed in the code
+- [ ] The master script tracks which scripts create and use which files
+- [ ] The data sets created by the reviewer are exactly the same as those shared by the coder
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:16]:
```suggestion
- [ ] The datasets created by the reviewer are exactly the same as those shared by the coder
```

Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md]:
```suggestion
- [ ] The datasets created by the reviewer are exactly the same as those shared by the coder
```

File: dime-coding-standards/checklists/construction-code-review.md [hunk 30]
```diff
@@ -0,0 +1,47 @@
...
+- [ ] Is there always clear documentation of why observations are dropped, if any?
+
+## Constructed dataset(s)
+- [ ] Is the resulting dataset tidy (each row is an observation, each column is a variable)?
+- [ ] Are variable names informative?
+- [ ] Are variable labels informative?
+- [ ] Are value labels informative?
+- [ ] Are all labels grammatically correct and free of special characters?
+- [ ] Is there clear documentation (variable dictionary, variable labels, value labels, notes, comments) about variable definition?
+- [ ] Are data set file names informative?
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/construction-code-review.md:47]:
```suggestion
- [ ] Are dataset file names informative?
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+## Code organization and readability
+- [ ] Code file names are informative 
+- [ ] It is clear in the code why tasks are being executed
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Clean data set checks (pre-publication)
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:27]:
```suggestion
## Clean dataset checks (pre-publication)
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It is clear in the code why tasks are being executed
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Clean dataset checks (pre-publication)
+- [ ] The data does not include direct identifiers
+- [ ] The data set has a clearly labeled, uniquely and fully identifying ID variable(s)
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:29]:
```suggestion
- [ ] The dataset has a clearly labeled, uniquely and fully identifying ID variable(s)
```

File: dime-coding-standards/checklists/data-cleaning-code-review.md [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] The code structure facilitates understanding of the tasks
+- [ ] Code uses white space to improve readability
+- [ ] There is extensive use of comments to explain the code
+- [ ] The code is efficient (tasks are executed in the simplest way possible, loops are used when needed rather than repeating lines, pre-defined functions are used)
+- [ ] Common tasks are abstracted and automated (e.g. using functions or macros)
+
+## Clean dataset checks (pre-publication)
+- [ ] The data does not include direct identifiers
+- [ ] The dataset has a clearly labeled, uniquely and fully identifying ID variable(s)
+- [ ] The level of observation of the data set is clear from the dataset name, ID variables, and documentation
```
Comment (luizaandrade) [inline dime-coding-standards/checklists/data-cleaning-code-review.md:30]:
```suggestion
- [ ] The level of observation of the dataset is clear from the dataset name, ID variables, and documentation
```

**Comment by bbdaniels** [comment]
OK this is good from me! Let's get it merged up? Others want to take a look?

**Pull request merged**